src/unit.rs

```diff
@@ -44,19 +44,11 @@ impl Unit {
             // otherwise, no chunking.
             .unwrap_or(false);
 
-        let is_secure = url.scheme().eq_ignore_ascii_case("https");
-
-        let hostname = url.host_str().unwrap_or(DEFAULT_HOST).to_string();
-
         let query_string = combine_query(&url, &req.query, mix_queries);
 
-        let cookie_headers: Vec<_> = {
-            let state = req.agent.lock().unwrap();
-            match state.as_ref().map(|state| &state.jar) {
-                None => vec![],
-                Some(jar) => match_cookies(jar, &hostname, url.path(), is_secure),
-            }
-        };
+        let state = req.agent.lock().unwrap();
```

Comment:
Instead of locking the mutex here, I would prefer if you pass the state into `extract_cookies()` and keep the lock local to there. It would alter semantics more than you have, but I don't think we need state to be locked until end of this function.

Reply:
Hey! Sorry for the lateness, but I've implemented this 👍